### PR TITLE
changed needed for  Demo

### PR DESCRIPTION
**Pull request by somnath77**
1. Rules get api at candidate side
2. Changed drive time left api
3. Changed candidate test time left api
4. Changes in candidate results serializer
5. Storing submission_count in problems table

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -31,13 +31,17 @@
...
+      get 'drives/:token', to: 'drives#show'
       resources :drives_candidates, only: [:update]
 
       resources :executions do
         post :submission_token, on: :collection
         get :submission_status, on: :member
       end
 
+      # resources :rules, only: %i[index]
+      get '/drives/:drive_id/rules', to: 'rules#index'
```
Comment (samu80) [inline config/routes.rb:43]: you can write like
resources :drives do
  resources :rules, only: [:index]
end

Comment (somnath77) [reply config/routes.rb:43]: it's becoming /drives/:drife_id/rules which is showing errors at frontend.

**Review by samu80 — APPROVED** [review]

File: app/controllers/api/v1/drives_controller.rb [hunk 2]
```diff
@@ -7,23 +7,24 @@ class DrivesController < ApiController
...
-        drives_candidate = DrivesCandidate.find_by(token: params[:id])
-        return render_error(message: I18n.t('token_not_found.message'), status: :not_found) unless drives_candidate
+        drive_candidate = DrivesCandidate.find_by(token: params[:id])
+        return render_error(message: I18n.t('drive_not_found.message'), status: :not_found) if drive_candidate.blank?
 
-        drive = Drive.find(drives_candidate.drive_id)
-        render_success(data: { drive: serialize_resource(drive, DriveSerializer) }, message: I18n.t('ok.message'))
+        drive = Drive.find_by(id: drive_candidate.drive_id)
+        if drive.present?
+          render json: { data: drive, candidate_id: drive_candidate.candidate_id, message: I18n.t('ok.message') }, status: 200
```
Comment (rahulknojha) [inline app/controllers/api/v1/drives_controller.rb:15]: why we are no using serializer here ?, can we use render_success method ?

File: app/controllers/api/v1/submissions_controller.rb [hunk 3]
```diff
@@ -6,9 +6,10 @@ class SubmissionsController < ApiController
       before_action :find_data
       def create
         if @drives_candidate.submissions.count < @submission_count
```
Comment (rahulknojha) [inline app/controllers/api/v1/submissions_controller.rb:8]: not sure why we are doing this ? we can always get submission time on run time by drives_candidate.submissions.count

File: config/initializers/devise_token_auth.rb [hunk 4]
```diff
@@ -18,7 +18,7 @@
 
   # Sets the max number of concurrent devices per user, which is 10 by default.
   # After this limit is reached, the oldest tokens will be removed.
-  # config.max_number_of_devices = 10
+  config.max_number_of_devices = 100
```
Comment (rahulknojha) [inline config/initializers/devise_token_auth.rb:21]: why this change needed?

File: config/initializers/devise_token_auth.rb [hunk 5]
```diff
@@ -51,7 +51,7 @@
   # By default, only Bearer Token authentication is implemented out of the box.
   # If, however, you wish to integrate with legacy Devise authentication, you can
   # do so by enabling this flag. NOTE: This feature is highly experimental!
-  config.enable_standard_devise_support = true
+  config.enable_standard_devise_support = false
```
Comment (rahulknojha) [inline config/initializers/devise_token_auth.rb:54]: why ?

**Review by rahulknojha — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/drives_controller.rb (hunk 2 above)
Comment (somnath77) [reply app/controllers/api/v1/drives_controller.rb:15]: candidate_id need to send with drive details which is not part of model, that's why didn't used serializer, If made changes in serializer, other controllers will show error

File: config/initializers/devise_token_auth.rb (hunk 4 above)
Comment (somnath77) [reply config/initializers/devise_token_auth.rb:21]: Removed

File: config/initializers/devise_token_auth.rb (hunk 5 above)
Comment (somnath77) [reply config/initializers/devise_token_auth.rb:54]: Reverted change

File: app/controllers/api/v1/drives_controller.rb (hunk 2 above)
Comment (somnath77) [reply app/controllers/api/v1/drives_controller.rb]: used render_success method

**Review by rahulknojha — APPROVED** [review]

**Pull request merged**